### PR TITLE
Add pulp-manifest utility from Pulp 2

### PR DESCRIPTION
**Pull request by dralley**
[noissue]

### CODE REVIEW

**Comment by ipanova** [comment]
since this pr has some 100+lines i would rather recommend opening an issue for it

**Comment by dralley** [comment]
@ipanova Added

File: pulp_manifest/pulp_manifest/build_manifest.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "directory",
+        help="A path to the directory where the PULP_MANIFEST" " file should be created.",
+    )
+    args = parser.parse_args()
+    directory = os.path.abspath(args.directory)
+
+    # Remove PULP_MANIFEST if it already exists in the given directory
```
Comment (mdellweg) [inline pulp_manifest/pulp_manifest/build_manifest.py:19]: Can't you just open the file in truncating mode?

File: pulp_manifest/test/unit/test_build_manifest.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        """Test getting the digest of a file."""
+        digest = build_manifest.get_digest(os.path.join(DATA_DIR, "a.txt"))
+        self.assertEquals(
+            "a1028f793b0aae9c51fa83e39975b254d78947620868f09e4a648e73486a623c", digest
+        )
+
+    def test_traverse_dir(self):
+        """Test directory traversal."""
+        manifest = build_manifest.traverse_dir(DATA_DIR)
+        self.assertEquals(len(manifest), 2)
```
Comment (mdellweg) [inline pulp_manifest/test/unit/test_build_manifest.py:21]: Shouldn't we test the `main` method of the `build_manifest` module? It contains most of the code.

File: pulp_manifest/test/data/b/c.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+This is another test file
```
Comment (mdellweg) [inline pulp_manifest/test/data/b/c.txt:1]: I'd prefer not to include these files, but generate them in a tempdir at the test setup stage.
But this is purely taste. You might ignore this suggestion.

**Review by mdellweg — COMMENTED** [review]
I agree, that this absolutely needs a changelog. Not judged by the number of lines, but the fact, that we ship it as a tool for the user (the console hook in setup.py).

File: pulp_manifest/test/unit/test_build_manifest.py (hunk 2 above)
Comment (dralley) [reply pulp_manifest/test/unit/test_build_manifest.py:21]: In theory, but it's not really that useful an exercise considering it's such a simple utility with a history of being used.

Comment (mdellweg) [reply pulp_manifest/test/unit/test_build_manifest.py:21]: That doesn't sound like a reason to have these tests either. ;)

File: pyproject.toml [hunk 4]
```diff
@@ -40,10 +40,9 @@ ignore = [
     "docs/**",
     "flake8.cfg",
     "template_config.yml",
-    ".travis/**",
```
Comment (ipanova) [inline pyproject.toml:43]: :+1:

**Review by ipanova — APPROVED** [review]

File: pulp_manifest/setup.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from setuptools import setup, find_packages
+
+setup(
+    name="pulp_manifest",
+    version="2.21a1",
```
Comment (ipanova) [inline pulp_manifest/setup.py:5]: do you want to change the version here?

**Comment by dralley** [comment]
@mdellweg I dropped the tests, didn't change the code though

@ipanova I changed the version to 3.0

File: pyproject.toml [hunk 6]
```diff
@@ -40,10 +40,9 @@ ignore = [
...
-    ".travis.yml",
     ".pep8speaks.yml",
     ".ci/**",
     ".github/**",
+    "pulp_manifest/**",
```
Comment (mdellweg) [inline pyproject.toml:46]: Not needed anymore...

Comment (dralley) [reply pyproject.toml:46]: Why is it not needed?  We don't want to ship these files with the actual pulp_file package.

Comment (mdellweg) [reply pyproject.toml:46]: Ah, i misread. This is a separate package in the same repo then. How do we plan to deliver it?

Comment (dralley) [reply pyproject.toml:46]: Probably pypi upload eventually.  For now I just wanted to put it in the repo for easier access.

**Pull request merged**